### PR TITLE
Adds Arch Linux installation guide

### PR DESCRIPTION
**Pull request by gdower**

### CODE REVIEW

**Comment by LocoDelAssembly** [comment]
@gdower Did the full test suite pass?

File: development/native/arch.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+Setup TaxonWorks project repository:
+
+```
+mkdir Projects
+cd Projects
+git clone https://github.com/SpeciesFileGroup/taxonworks.git
+cd taxonworks
+```
+
+When you do `cd taxonworks` you will see a message regarding a particular version of Ruby.  Install that version of Ruby with the command provided in the terminal. It will look something like: `rvm install 2.7.1`.
```
Comment (LocoDelAssembly) [inline development/native/arch.md]: Should be `2.7.2`

**Comment by gdower** [comment]
@LocoDelAssembly I'll fix the version. There were 8 test failures. Here's the output:

```
Finished in 26 minutes 21 seconds (files took 10.69 seconds to load)
7299 examples, 8 failures, 10 pending

Failed examples:

rspec ./spec/features/tasks/otus/filter_spec.rb:43 # tasks/otus/filter using simple_world signed in as a user #set_area renders count of otus in a drawn area corresponding to b
rspec ./spec/lib/vendor/serrano_spec.rb:63 # TaxonWorks::Vendor::Serrano#new_from_citation from DOI text http
rspec ./spec/lib/vendor/serrano_spec.rb:94 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  text1
rspec ./spec/lib/vendor/serrano_spec.rb:103 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  text2
rspec ./spec/lib/vendor/serrano_spec.rb:109 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  html1
rspec ./spec/lib/vendor/serrano_spec.rb:119 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  html2
rspec ./spec/models/sqed_depiction_spec.rb:56 # SqedDepiction #rebuild 1
rspec ./spec/models/sqed_depiction_spec.rb:61 # SqedDepiction #rebuild 
```

**Comment by LocoDelAssembly** [comment]
```
rspec ./spec/models/sqed_depiction_spec.rb:56 # SqedDepiction #rebuild 1
rspec ./spec/models/sqed_depiction_spec.rb:61 # SqedDepiction #rebuild 
```

These two I think are not one of those that randomly fail. Can you run them and paste the full output here?

**Comment by gdower** [comment]
Below is the full output. Maybe I need to configure tesseract and set the `TESSDATA_PREFIX` variable.

```
Failures:

  1) tasks/otus/filter using simple_world signed in as a user #set_area renders count of otus in a drawn area corresponding to b
     Failure/Error: expect(page).to have_css('#area_count', text: '4')
       expected to find visible css "#area_count" with text "4" but there were no matches. Also found "3", which matched the selector but not all filters.                                  
     # ./spec/features/tasks/otus/filter_spec.rb:54:in `block (5 levels) in <top (required)>'

  2) TaxonWorks::Vendor::Serrano#new_from_citation from DOI text http
     Failure/Error: expect(s.class).to eq(Source::Bibtex)
     
       expected: Source::Bibtex(id: integer, serial_id: integer, address: string, annote: string, booktitle: string, c...url: string, author: text, cached: text, cached_author_string: text, cached_nomenclature_date: date)                                                             
            got: Source::Verbatim(id: integer, serial_id: integer, address: string, annote: string, booktitle: string,...url: string, author: text, cached: text, cached_author_string: text, cached_nomenclature_date: date)                                                             
     
       (compared using ==)
     
       Diff:
       @@ -1 +1 @@
       -Source::Bibtex(id: integer, serial_id: integer, address: string, annote: string, booktitle: string, chapter: string, crossref: string, edition: string, editor: string, howpublished: string, institution: string, journal: string, key: string, month: string, note: string, number: string, organization: string, pages: string, publisher: string, school: string, series: string, title: text, type: string, volume: string, doi: string, abstract: text, copyright: text, language: string, stated_year: string, verbatim: string, created_at: datetime, updated_at: datetime, bibtex_type: string, created_by_id: integer, updated_by_id: integer, day: integer, year: integer, isbn: string, issn: string, verbatim_contents: text, verbatim_keywords: text, language_id: integer, translator: string, year_suffix: string, url: string, author: text, cached: text, cached_author_string: text, cached_nomenclature_date: date)                             
       +Source::Verbatim(id: integer, serial_id: integer, address: string, annote: string, booktitle: string, chapter: string, crossref: string, edition: string, editor: string, howpublished: string, institution: string, journal: string, key: string, month: string, note: string, number: string, organization: string, pages: string, publisher: string, school: string, series: string, title: text, type: string, volume: string, doi: string, abstract: text, copyright: text, language: string, stated_year: string, verbatim: string, created_at: datetime, updated_at: datetime, bibtex_type: string, created_by_id: integer, updated_by_id: integer, day: integer, year: integer, isbn: string, issn: string, verbatim_contents: text, verbatim_keywords: text, language_id: integer, translator: string, year_suffix: string, url: string, author: text, cached: text, cached_author_string: text, cached_nomenclature_date: date)                           
       
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/vcr-5.1.0/lib/vcr/util/variable_args_block_caller.rb:9:in `call_block'                                                                           
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/vcr-5.1.0/lib/vcr.rb:188:in `use_cassette'

  3) TaxonWorks::Vendor::Serrano#cached_string correctly formats  text1
     Failure/Error: expect(src1.cached_string('text')).to start_with('Brauer, A. (1909) Die Süsswasserfauna Deutschlands. Eine Exkursionsfauna bearb. ... und hrsg. von Dr. Brauer. G. Fischer, Available from: https://doi.org/10.5962%2Fbhl.title.1086')
     
     NoMethodError:
       undefined method `cached_string' for #<Source::Verbatim:0x000055960fea0678>
       Did you mean?  cached_author_string
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activemodel-6.0.3.3/lib/active_model/attribute_methods.rb:432:in `method_missing'                                                                

  4) TaxonWorks::Vendor::Serrano#cached_string correctly formats  text2
     Failure/Error:
       expect(src2.cached_string('text')).to eq(
         'Kevan, D.K.M.E. & Wighton, D.C. (1981) Paleocene orthopteroids from south-central Alberta, Canada. Canadian Journal of Earth Sciences 18, 1824–1837.'
       )
     
     NoMethodError:
       undefined method `cached_string' for #<Source::Verbatim:0x000055960e022fe8>
       Did you mean?  cached_author_string
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activemodel-6.0.3.3/lib/active_model/attribute_methods.rb:432:in `method_missing'                                                                

  5) TaxonWorks::Vendor::Serrano#cached_string correctly formats  html1
     Failure/Error:
       expect(src1.cached_string('html')).to start_with(
         'Brauer, A. (1909) <i>Die Süsswasserfauna Deutschlands. Eine Exkursionsfauna bearb. ... und hrsg. von Dr. Brauer.</i> G. Fischer, Available from: https://doi.org/10.5962%2Fbhl.title.1086')
     
     NoMethodError:
       undefined method `cached_string' for #<Source::Verbatim:0x000055961d51e218>
       Did you mean?  cached_author_string
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activemodel-6.0.3.3/lib/active_model/attribute_methods.rb:432:in `method_missing'                                                                

  6) TaxonWorks::Vendor::Serrano#cached_string correctly formats  html2
     Failure/Error:
       expect(src2.cached_string('html')).to eq(
         'Kevan, D.K.M.E. &amp; Wighton, D.C. (1981) Paleocene orthopteroids from south-central Alberta, Canada. <i>Canadian Journal of Earth Sciences</i> 18, 1824–1837.'
       )
     
     NoMethodError:
       undefined method `cached_string' for #<Source::Verbatim:0x000055961d2c61f0>
       Did you mean?  cached_author_string
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activemodel-6.0.3.3/lib/active_model/attribute_methods.rb:432:in `method_missing'                                                                

  7) SqedDepiction #rebuild 1
     Failure/Error: rescue NoMethodError # TODO - better handling in Sqed
     
     TypeError:
       class or module required for rescue clause
     # ./app/models/sqed_depiction.rb:176:in `preprocess'
     # ./app/models/sqed_depiction.rb:63:in `recalculate'
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:428:in `block in make_lambda'                                                              
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:238:in `block in halting_and_conditional'                                                  
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:517:in `block in invoke_after'                                                             
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:517:in `each'                                                                              
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:517:in `invoke_after'                                                                      
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:136:in `run_callbacks'                                                                     
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:825:in `_run_save_callbacks'                                                               
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/callbacks.rb:327:in `create_or_update'                                                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/timestamp.rb:128:in `create_or_update'                                                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/persistence.rb:503:in `save!'                                                                             
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/validations.rb:53:in `save!'                                                                              
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:318:in `block in save!'                                                                   
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:375:in `block in with_transaction_returning_status'                                       
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `block in transaction'                         
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/transaction.rb:280:in `block in within_new_transaction'                      
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'                     
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'                                
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'                                         
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/transaction.rb:278:in `within_new_transaction'                               
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `transaction'                                  
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:212:in `transaction'                                                                      
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:366:in `with_transaction_returning_status'                                                
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:318:in `save!'                                                                            
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/suppressor.rb:48:in `save!'                                                                               
     # ./spec/models/sqed_depiction_spec.rb:58:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RTesseract::Error:
     #   Error opening data file /usr/share/tessdata/eng.traineddata
     #   Please make sure the TESSDATA_PREFIX environment variable is set to your "tessdata" directory.                                                                                     
     #   Failed loading language 'eng'
     #   Tesseract couldn't load any languages!
     #   Could not initialize tesseract.
     #   /home/gdo/.rvm/gems/ruby-2.7.2/gems/rtesseract-3.1.2/lib/rtesseract/command.rb:57:in `run'                                                                                         

  8) SqedDepiction #rebuild 2
     Failure/Error: rescue NoMethodError # TODO - better handling in Sqed
     
     TypeError:
       class or module required for rescue clause
     # ./app/models/sqed_depiction.rb:176:in `preprocess'
     # ./app/models/sqed_depiction.rb:63:in `recalculate'
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:428:in `block in make_lambda'                                                              
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:238:in `block in halting_and_conditional'                                                  
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:517:in `block in invoke_after'                                                             
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:517:in `each'                                                                              
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:517:in `invoke_after'                                                                      
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:136:in `run_callbacks'                                                                     
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:825:in `_run_save_callbacks'                                                               
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/callbacks.rb:327:in `create_or_update'                                                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/timestamp.rb:128:in `create_or_update'                                                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/persistence.rb:470:in `save'                                                                              
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/validations.rb:47:in `save'                                                                               
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:314:in `block in save'                                                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:375:in `block in with_transaction_returning_status'                                       
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/database_statements.rb:278:in `transaction'                                  
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:212:in `transaction'                                                                      
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:366:in `with_transaction_returning_status'                                                
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:314:in `save'                                                                             
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/suppressor.rb:44:in `save'                                                                                
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/persistence.rb:621:in `block in update'                                                                   
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:375:in `block in with_transaction_returning_status'                                       
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `block in transaction'                         
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/transaction.rb:280:in `block in within_new_transaction'                      
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'                     
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'                                
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'                                    
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activesupport-6.0.3.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'                                         
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/transaction.rb:278:in `within_new_transaction'
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `transaction'
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:212:in `transaction'
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:366:in `with_transaction_returning_status'
     # /home/gdo/.rvm/gems/ruby-2.7.2/gems/activerecord-6.0.3.3/lib/active_record/persistence.rb:619:in `update'
     # ./spec/models/sqed_depiction_spec.rb:62:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RTesseract::Error:
     #   Error opening data file /usr/share/tessdata/eng.traineddata
     #   Please make sure the TESSDATA_PREFIX environment variable is set to your "tessdata" directory.
     #   Failed loading language 'eng'
     #   Tesseract couldn't load any languages!
     #   Could not initialize tesseract.
     #   /home/gdo/.rvm/gems/ruby-2.7.2/gems/rtesseract-3.1.2/lib/rtesseract/command.rb:57:in `run'

Finished in 26 minutes 21 seconds (files took 10.69 seconds to load)
7299 examples, 8 failures, 10 pending

Failed examples:

rspec ./spec/features/tasks/otus/filter_spec.rb:43 # tasks/otus/filter using simple_world signed in as a user #set_area renders count of otus in a drawn area corresponding to b
rspec ./spec/lib/vendor/serrano_spec.rb:63 # TaxonWorks::Vendor::Serrano#new_from_citation from DOI text http
rspec ./spec/lib/vendor/serrano_spec.rb:94 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  text1
rspec ./spec/lib/vendor/serrano_spec.rb:103 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  text2
rspec ./spec/lib/vendor/serrano_spec.rb:109 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  html1
rspec ./spec/lib/vendor/serrano_spec.rb:119 # TaxonWorks::Vendor::Serrano#cached_string correctly formats  html2
rspec ./spec/models/sqed_depiction_spec.rb:56 # SqedDepiction #rebuild 1
rspec ./spec/models/sqed_depiction_spec.rb:61 # SqedDepiction #rebuild 2
```

**Comment by gdower** [comment]
@LocoDelAssembly  Installing the `tesseract-data-eng` package fixed the failed tesseract tests. I modified the instructions to include installing any desired tesseract language packages.

**Comment by LocoDelAssembly** [comment]
Only remaining failing specs are those using `Serrano`? If so then we are ready to merge.

**Comment by gdower** [comment]
There is one other failed test for the filter OTU tests: `renders count of otus in a drawn area corresponding to b (FAILED - 1)`. Selenium pauses on that test for a while after drawing the filter box on the map, and then it fails. Is that typical?

```
rspec ./spec/features/tasks/otus/filter_spec.rb
DEPRECATION WARNING: ActionView::Base instances should be constructed with a lookup context, assignments, and a controller. (called from <top (required)> at /home/gdo/taxonworks/config/environment.rb:5)
Run options: exclude {:lint=>true}
Faker random seed: 309422743764219154232244253907694474765

tasks/otus/filter
  using simple_world
    behaves like a_login_required_and_project_selected_controller
      authorization
        submitting request without sign_in redirects to root_path
          is expected to have visible button "Sign in" that is not disabled
        submitting a request with sign_in but no project redirects to dashboard
          is expected to have text "Dashboard"
        submitting a request with sign_in and project selected renders
          is expected to have text "Otus by area"
    signed in as a user
      #set_area
        renders count of otus in a specific names area A
        renders count of otus in a specific names area B
        renders count of otus in a specific names area E
        renders count of otus in a drawn area corresponding to b (FAILED - 1)
        renders count of otus in a drawn area
      #set_nomen
        renders count of otus from a specific name without descendants
        renders count of otus from a taxon name of its own rank with descendants
        renders count of otus from a specific name with descendants without rank specified
        renders count of otus from a specific name with descendants with specific rank
      #set_author
        selects a taxon name author list (and)
        selects a taxon name author list (or)
      #find
        renders count of objects and table found using a drawn area and date range
      #find none
        renders count of objects and table found using name

Failures:

  1) tasks/otus/filter using simple_world signed in as a user #set_area renders count of otus in a drawn area corresponding to b
     Failure/Error: expect(page).to have_css('#area_count', text: '4')
       expected to find visible css "#area_count" with text "4" but there were no matches. Also found "3", which matched the selector but not all filters. 
     # ./spec/features/tasks/otus/filter_spec.rb:54:in `block (5 levels) in <top (required)>'

Finished in 2 minutes 20 seconds (files took 8.99 seconds to load)
16 examples, 1 failure

Failed examples:

rspec ./spec/features/tasks/otus/filter_spec.rb:43 # tasks/otus/filter using simple_world signed in as a user #set_area renders count of otus in a drawn area corresponding to b
```

**Comment by LocoDelAssembly** [comment]
@gdower take a look at the video attached. That is what should happen.

[filter_spec.zip](https://github.com/SpeciesFileGroup/install_taxonworks/files/5639685/filter_spec.zip)

At which point does it stop at your end?

**Comment by gdower** [comment]
@LocoDelAssembly, it pauses for a minute at this step:

image

Then fails with:

image

Which I think corresponds to this test:

```ruby
        it 'renders count of otus in a drawn area corresponding to b' do
          visit(index_path)
          find('#label_toggle_slide_area').click
          # TODO: Helper method to calculate x-y and perform interaction
          find('#drawable_map_canvas')
            .hover.click(x: 256, y: 128) # [  0,   0]
            .hover.click(x: 270, y: 128) # [ 10,   0]
            .hover.click(x: 270, y: 143) # [ 10, -10]
            .hover.click(x: 256, y: 143) # [  0, -10]
            .hover.click(x: 256, y: 128) # [  0,   0]
          click_button('Set area')
          expect(page).to have_css('#area_count', text: '4')
        end
```

So I think it's expecting 4 and gets 3 otus?

**Comment by LocoDelAssembly** [comment]
If that is the case it might be an issue with Postgres/PostGIS.

**Comment by LocoDelAssembly** [comment]
What this shows you?
```
$ psql -U taxonworks_development
psql (10.15 (Ubuntu 10.15-1.pgdg18.04+1))
Type "help" for help.

taxonworks_development=# SELECT PostGIS_Version();
            postgis_version            
---------------------------------------
 2.4 USE_GEOS=1 USE_PROJ=1 USE_STATS=1
(1 row)
```
(My current setup)

```
$ psql -U taxonworks_development
psql (12.4 (Ubuntu 12.4-1.pgdg20.04+1))
Type "help" for help.

taxonworks_development=# SELECT PostGIS_Version();
            postgis_version            
---------------------------------------
 3.0 USE_GEOS=1 USE_PROJ=1 USE_STATS=1
(1 row)
```
(Ubuntu 20.04 VM, spec also passing)

**Comment by gdower** [comment]
If I change that test to expect 3, it does pass.

Here's my postgis version:
```
psql -U taxonworks_development
psql (12.5)
Type "help" for help.

taxonworks_development=# SELECT PostGIS_Version();
            postgis_version            
---------------------------------------
 3.0 USE_GEOS=1 USE_PROJ=1 USE_STATS=1
(1 row)
```

**Comment by gdower** [comment]
@LocoDelAssembly, I noticed that if I maximize the selenium firefox browser window that the test passes, so I think selenium is clicking the wrong coordinates when drawing the map filter box if the window isn't maximized.

**Comment by LocoDelAssembly** [comment]
Perfect @gdower I think it all OK then. Merging. Thanks for all the work!

**Pull request merged**